include/operations/blas_constants.h

```diff
@@ -58,71 +60,66 @@ enum class const_val : int {
   m_two = -2,
   max = 3,
   min = 4,
-  imax = 5,
-  imin = 6
-};
-/*!
-@def define a specialization of the constant template value for each indicator.
-@ref ConstValue.
-@tparam primitive_t The value type to specialize for.
-@tparam Indicator The constant to specialize for.
-*/
-template <typename primitive_t, const_val Indicator>
-struct ConstValue {
-  constexpr static SYCL_BLAS_INLINE primitive_t init() {
-    return static_cast<primitive_t>(static_cast<int>(Indicator));
-  }
-};
-template <typename primitive_t>
-struct ConstValue<primitive_t, const_val::max> {
-  constexpr static SYCL_BLAS_INLINE primitive_t init() {
-    return std::numeric_limits<primitive_t>::min();
-  }
-};
-
-template <typename primitive_t>
-struct ConstValue<primitive_t, const_val::min> {
-  constexpr static SYCL_BLAS_INLINE primitive_t init() {
-    return std::numeric_limits<primitive_t>::max();
-  }
+  abs_max = 5,
+  abs_min = 6,
 };
 
 /*!
 @brief Template struct used to represent constants within a compile-time
-expression tree, each instantiation will have a static constexpr member variable
-of the type value_t initialized to the specified constant.
+expression tree, each instantiation will have a static constexpr member
+variable of the type value_t initialized to the specified constant.
 @tparam value_t Value type of the constant.
 @tparam kIndicator Enumeration specifying the constant.
 */
 template <typename value_t, const_val Indicator>
 struct constant {
   constexpr static SYCL_BLAS_INLINE value_t value() {
-    return ConstValue<value_t, Indicator>::init();
+    return static_cast<value_t>(Indicator);
+  }
+};
+
+template <typename value_t>
+struct constant<value_t, const_val::max> {
+  constexpr static SYCL_BLAS_INLINE value_t value() {
+    return std::numeric_limits<value_t>::max();
   }
 };
 
-template <typename value_t, typename index_t>
-struct constant<IndexValueTuple<value_t, index_t>, const_val::imax> {
-  constexpr static SYCL_BLAS_INLINE IndexValueTuple<value_t, index_t> value() {
-    return IndexValueTuple<value_t, index_t>(
-        std::numeric_limits<index_t>::max(),
-        static_cast<value_t>(0)); // This is used for absolute max, -1 == 1
+template <typename value_t>
+struct constant<value_t, const_val::min> {
+  constexpr static SYCL_BLAS_INLINE value_t value() {
+    return std::numeric_limits<value_t>::min();
   }
 };
 
-template <typename value_t, typename index_t>
-struct constant<IndexValueTuple<value_t, index_t>, const_val::imin> {
-  constexpr static SYCL_BLAS_INLINE IndexValueTuple<value_t, index_t> value() {
-    return IndexValueTuple<value_t, index_t>(
-        std::numeric_limits<index_t>::max(),
-        std::numeric_limits<value_t>::max());
+template <typename value_t>
+struct constant<value_t, const_val::abs_max> {
+  constexpr static SYCL_BLAS_INLINE value_t value() {
+    return std::numeric_limits<value_t>::max();
```

Comment:
As discussed offline, it might make sense to combine that with the specialization for `max`, which I think could be done with `enable_if`, if it doesn't make the code less readable.